### PR TITLE
[oneDNN] Fix to 34554 (same as previous PR but should build with GPU)

### PR DESCRIPTION
**Pull request by jczaja**
### PR types
Bug fixes

### PR changes
OPs 

### Describe
Same as #34623 but should build with GPU

### CODE REVIEW

**Review by sfraczek — APPROVED** [review]
LGTM

**Comment by jczaja** [comment]
@chenwhql This PR is the same #34623 + some changes to have UT build for GPU builds. Cuould you please test if this builds for GPU build of PaddlePaddle (scenario where #34623 failed) ?

**Comment by jczaja** [comment]
@OliverLPH I got more models so I can test your commandline in regard of #34554 :
`./build/model_test --single_thread=false --single_instance=false --test_groups="3"`

It works fine for oneDNN enabled (using this PR) , but it does crash when oneDNN is not enabled (using this PR).
Could you please make sure that when testing you have oneDNN enabled e.g.
In model_test.cc (line 90) 
_config.EnableMKLDNN();_
is enabled?

File: paddle/fluid/operators/mkldnn/caching_tests.cmake [hunk 1]
```diff
@@ -1 +1,6 @@
-cc_test(test_mkldnn_caching SRCS mkldnn/test_mkldnn_caching.cc DEPS op_registry elementwise_mul_op elementwise_add_op activation_op softmax_op softmax scope device_context enforce)
+set(TEST_MKLDNN_CACHING_DEPS op_registry elementwise_mul_op elementwise_add_op activation_op softmax_op conv_op im2col vol2col softmax scope device_context enforce)
+if (WITH_GPU OR WITH_ROCM)
+  set(TEST_MKLDNN_CACHING_DEPS ${TEST_MKLDNN_CACHING_DEPS} depthwise_conv)
+endif()
+cc_test(test_mkldnn_caching SRCS mkldnn/test_mkldnn_caching.cc DEPS ${TEST_MKLDNN_CACHING_DEPS})
```
Comment (lidanqing-vv) [inline paddle/fluid/operators/mkldnn/caching_tests.cmake:5]: I wonder why depthwise_conv have to be in DEPS while the test did not use it?
If it is really needed, what about always put depthwise_conv always as one of the DEPS. Cause I think Baidu has other hardware too like kunlun.

**Comment by OliverLPH** [comment]
@jczaja thanks, I will test all models with oneDNN enabled

File: paddle/fluid/operators/mkldnn/caching_tests.cmake (hunk 1 above)
Comment (jczaja) [reply paddle/fluid/operators/mkldnn/caching_tests.cmake:5]: conv on GPU got depedepndency on depthwise_conv. Without it there is linking error: unresolved DepthwiseConv.... So i had to add it

**Review by lidanqing-vv — APPROVED** [review]
LGTM

**Comment by jczaja** [comment]
@chenwhql  Could you please approve PR-CI-APPROVAL as the failure is exactly the same as then on in #34623 that you have approved?

**Comment by lidanqing-vv** [comment]
@chenwhql Could you please approve this PR again, changes are almost same as previous PR you approved. 
About muti-threading issue, @OliverLPH  confirmed that even with mkldnn OFF, those tests also fail so it is not mkldnn integration issues.
So you may approve this PR without multi-threading concern, thanks !

**Review by chenwhql — APPROVED** [review]
LGTM for PADDLE_ENFORCE

**Comment by baoachun** [comment]
> @OliverLPH I got more models so I can test your commandline in regard of #34554 :
> `./build/model_test --single_thread=false --single_instance=false --test_groups="3"`
> 
> It works fine for oneDNN enabled (using this PR) , but it does crash when oneDNN is not enabled (using this PR).
> Could you please make sure that when testing you have oneDNN enabled e.g.
> In model_test.cc (line 90)
> _config.EnableMKLDNN();_
> is enabled?

Hi, I have re-executed the test program, and I found that there will be dnnl errors when config.mkldnn_enabled() is 0. Here is the log [without_mkldnn.log](https://github.com/PaddlePaddle/Paddle/files/6998222/without_mkldnn.log).

**Pull request merged**